Jjdzr11 ko 25

KozlowskiKamil opened the pull request.

Commit:
dostosowanie menu do wczytania użytkownika, refaktor kodu klasy BorrowBooks w metodzie addBookToPerson()

Commit:
refaktor kodu klasy BorrowBooks w metodzie returnBook()

Commit:
poprawiona metoda zwracania książek

Commit:
rozbudowa menu bibliotekarza o możliwość wyświetlania listy wypożyczonych książek oraz listę dostępnych książek

Commit:
changed menu

KozlowskiKamil committed:
ID ok

KozlowskiKamil committed:
ID ok, logowanie login /hasło -ok

KozlowskiKamil committed:
ID ok, logowanie login /hasło -ok do poprawy kosmetyka

KozlowskiKamil committed:
ID ok, logowanie login /hasło -ok do poprawy kosmetyka + małe sprzątanie

KozlowskiKamil committed:
ID ok, logowanie login /hasło -ok do poprawy kosmetyka + małe sprzątanie.
Dodanie paskage

KozlowskiKamil committed:
Dodanie weryfikacji czy istnieje użytkownik o podanym loginie podczas rejestracji nowego usera.

KozlowskiKamil committed:
Poprawienie logowania bibliotekarza na bardziej uniwersalne.
Gotowe do PR

KozlowskiKamil committed:
Poprawa buga z second name

KozlowskiKamil committed:
Poprawa literówki logi -> login

TomaszOdwald commented:
Na pewno usuń "Class" z package'y, niepotrzebne to

TomaszOdwald commented on src/main/java/pl/isa/biblioteka/booksClass/BorrowBooks.java:
```diff
@@ -10,70 +10,44 @@ public class BorrowBooks {
+            if (book.getTitle().equalsIgnoreCase(bookTitle) && book.isState()) {
+                book.setState(false);
+                personBooks.add(book);
+                System.out.println("Książka została wypożyczona");
```
Metoda mogłaby zwracać, czy udało się wypożyczyć książki albo jakich się nie udało, itp. Ale musielibyście stworzyć osobną klasę, która zwracałaby rezultat, więc niech tak zostanie ;)

TomaszOdwald commented on src/main/java/pl/isa/biblioteka/booksClass/BorrowBooks.java:
```diff
@@ -10,70 +10,44 @@ public class BorrowBooks {
         }
-        if (!findUser) {
-            System.out.println("Brak Użytkownika o podanych danych");
+        if (!findBook) {
```
Ale skoro i tak macie booleana, który po prostu wypisuje czy się udało czy nie, więc można go zwrócić i w miejscu, w którym wywołujesz addBookToPerson sprawdzić, co ta metoda zwróciła

Wenderholm replied:
przeniesono do menu gdzie wyświetlana jest metoda

TomaszOdwald commented on src/main/java/pl/isa/biblioteka/booksClass/BorrowBooks.java:
```diff
@@ -10,70 +10,44 @@ public class BorrowBooks {
         }
-        if (!findUser) {
-            System.out.println("Brak Użytkownika o podanych danych");
+        if (!findBook) {
```
Podobnie jak wyżej, można tu zwrócić booleana i pozbędziesz się wypisywania czegokolwiek w konsoli w tej klasie

Wenderholm replied:
przeniesono do menu gdzie wyświetlana jest metoda

KozlowskiKamil committed:
Usunięcie dopisku class z package

TomaszOdwald commented on src/main/java/pl/isa/biblioteka/booksClass/BorrowBooks.java:
```diff
@@ -10,70 +10,44 @@ public class BorrowBooks {
+        for (Book book : booksList) {
+            if (book.getTitle().equalsIgnoreCase(bookTitle) && book.isState()) {
+                book.setState(false);
+                personBooks.add(book);
```
To teraz trochę skomplikuje, bo zamiast przekazywać listę książek, przekazuj nazwę, jaką wprowadził użytkownik (patrz linia ze Scannerem). Natomiast jeśli mamy zalogowanego użytkownika, to mamy też zapewne metodę, która potrafi nam go zwrócić, a wtedy już prosta droga do tego, żeby przypisać mu książki

TomaszOdwald commented on src/main/java/pl/isa/biblioteka/booksClass/BorrowBooks.java:
```diff
@@ -108,28 +82,18 @@ private static void showFilterBookByCategory(List<Book> books, Set<String> avail
     }
 
-    public void mainLoop() {
+    public void mainLoop(List<Book> personBooks) {
```
Tu podobnie, nie przez parametr, a przez odczytywanie zalogowanego użytkownika ;)

Wenderholm replied:
stworzono klasę przetrzymującą zalogowanego użytkownika

TomaszOdwald commented on src/main/java/pl/isa/biblioteka/userClass/Person.java:
```diff
@@ -1,67 +1,85 @@
+    public void showPersonBooks() {
         for (Book personBook : personBooks) {
-            System.out.printf("Tytuł: %s, Autor: %s %n", personBook.getTitle(),personBook.getAuthor());
+            System.out.printf("Tytuł: %s, Autor: %s %n", personBook.getTitle(), personBook.getAuthor());
```
Żadnych printów w Person

Wenderholm replied:
usunięto tą część

KozlowskiKamil commented:
> Na pewno usuń "Class" z package'y, niepotrzebne to
Dopisek Class usunięty.

TomaszOdwald commented on src/main/java/pl/isa/biblioteka/userClass/Users.java:
W dużym skrócie, 2 osobne klasy z tego, 1 do zarządzania użytkownikami, a druga do wyświetlania menu i wczytywania danych

Commit:
- przeniesiono wypisywanie z metod addBooksToPerson() oraz returnBook() do menu odpowiedzialne za wyświetlanie tych metod
- usunięto parametry z metod
- wydzielono klasę przetrzymującą zalogowanego użytkownika - LogUser
- wydzielono klasę tworzącą menu dla wypożyczenia książki -  BooksBorrowedMenu

TomaszOdwald commented on src/main/java/pl/isa/biblioteka/books/BorrowBooks.java:
```diff
@@ -0,0 +1,84 @@
+
+    public static boolean addBookToPerson() {
+        List<Book> booksList = BooksEdit.booksList;
+        System.out.println("Podaj tytuł książki do wypożyczenia: ");
```
Dalej nie do końca to o czym myślałem, bookTitle powinno przychodzić jako parametr z menu, a nie być wczytywane tutaj.

Wenderholm replied:
przeniesione do menu

TomaszOdwald commented on src/main/java/pl/isa/biblioteka/books/BorrowBooks.java:
```diff
@@ -0,0 +1,84 @@
+    public static boolean returnBook() {
+        List<Book> booksList = BooksEdit.booksList;
+        System.out.println("Podaj tytuł książki do zwrócenia: ");
+        String bookTitleToReturn = scanner.nextLine();
```
j.w. - bookTitleToReturn jako parametr

Wenderholm replied:
przeniesione do menu

TomaszOdwald commented on src/main/java/pl/isa/biblioteka/books/BorrowBooks.java:
```diff
@@ -0,0 +1,84 @@
+        return availableCategory;
+    }
+
+    private static void showFilterBookByCategory(List<Book> books, Set<String> availableCategory) {
```
To powinno robić menu, a sortByCategory powinno zwrócić po prostu tą listę do menu.

TomaszOdwald commented on src/main/java/pl/isa/biblioteka/books/BorrowBooks.java:
```diff
@@ -0,0 +1,84 @@
+
+    }
+
+    private static Set<String> showAvailableCategory(List<Book> books) {
```
To też powinno robić menu, a BorrowBooks powinien tylko zwracać listę dostępnych kategorii.

TomaszOdwald commented on src/main/java/pl/isa/biblioteka/menu/BooksBorrowMenu.java:
```diff
@@ -0,0 +1,106 @@
+        LogUser logUser = new LogUser();
+        switch (option) {
+            case BORROW_BOOK -> {
+                if ((addBookToPerson())) {
```
Jako że tu powinno dojść wczytywanie, to fajnie żeby wszystkie case'y były tylko wywołaniem funkcji, czyli żeby nie było potrzeby korzystania z nawiasów { }

Wenderholm replied:
poprawiono. w menu znajdują się tylko wywołania

TomaszOdwald commented on src/main/java/pl/isa/biblioteka/menu/BooksBorrowMenu.java:
```diff
@@ -0,0 +1,106 @@
+    }
+
+    private void close() {
+        System.out.println("Bye Bye!");
```
Nie widzę co close() robi, żeby zamknąć program, poza wypisaniem tekstu do konsoli ;)

Wenderholm replied:
Usunięto tekst próbny

TomaszOdwald commented on src/main/java/pl/isa/biblioteka/menu/Menu.java:
```diff
@@ -0,0 +1,170 @@
+        }
+    }
+
+    protected void librarianMenu() {
```
Może warto zrobić z tego osobną klasę? ;)

Wenderholm replied:
dodano klasę LibrarianMenu gdzie przeniesiono menu bibliotekarza

TomaszOdwald commented on src/main/java/pl/isa/biblioteka/menu/Menu.java:
```diff
@@ -0,0 +1,170 @@
+        }
+    }
+
+    protected void userMenu(String login, String password) {
```
I tu podobnie ;)

Wenderholm replied:
dodano klasę UserMenu gdzie przeniesiono menu użytkownika

TomaszOdwald commented on src/main/java/pl/isa/biblioteka/menu/Menu.java:
```diff
@@ -0,0 +1,170 @@
+
+    protected void selectUser() {
+        List<Person> persons = Users.getUsers();
+        LogUser logUser = new LogUser();
```
Niepotrzebne jest tworzenie obiektu dla klasy, która udostępnia tylko statyczne funkcje.

Wenderholm replied:
usunięte

TomaszOdwald commented on src/main/java/pl/isa/biblioteka/menu/Menu.java:
```diff
@@ -0,0 +1,170 @@
+        System.out.println("Podaj swoje hasło");
+        String password = scanner.nextLine().trim();
+        boolean found = false;
+        for (Person person1 : persons) {
```
Logikę wczytywania użytkownika warto byłoby przenieść do tej samej klasy, co trzyma zalogowanego użytkownika, dzięki czemu mamy odseparowany kod do wyświetlania/wczytywania danych, od kodu do przeszukiwania bazy.

Wenderholm replied:
Logika została przeniesiona do LogUser

TomaszOdwald commented on src/main/java/pl/isa/biblioteka/user/LogUser.java:
```diff
@@ -0,0 +1,23 @@
+import java.util.List;
+
+public class LogUser {
+    private static Person logPerson;
```
Statycznej zmiennej niepotrzebne są gettery i settery, jeśli nie zawierają żadnej logiki, więc ta zmienna mogłaby być śmiało public.

Wenderholm replied:
poprawione

TomaszOdwald commented on src/main/java/pl/isa/biblioteka/user/PersonService.java:
W sumie to tu już macie idealną klasę, która może posłużyć do przechowywania zalogowanego użytkownika, niepotrzebnie tworzyliście nową ;)

TomaszOdwald commented on src/main/java/pl/isa/biblioteka/user/Users.java:
```diff
@@ -0,0 +1,80 @@
+        return users.removeIf(findByID(id));
+    }
+
+    public static Predicate<Person> findByID(Integer id) {
```
Takie rzeczy jak znajdywanie czy usuwanie powinien robić PersonService, bo teraz nie wiem czy klasa Users jest waszym menu czy controllerem, bo robi tego i tego po trochu ;)

Commit:
zmiana metody addBookToPerson

Commit:
zmiana metody returnBook

Commit:
refactor kodu

kingabiereg committed:
Dodanie metody wyszukiwania dostepnych ksiazek po autorze

kingabiereg committed:
Dodano metode wyszukiwania dostepnej ksiazki po tytule

KozlowskiKamil committed:
Czyszczenie importów plus drobne korekty w komunikatach

Commit:
refactor2
